### ISSUE TITLE
TSM1: Panic "missed stack barrier" when profiling enabled

### ISSUE THREAD

**Issue by otoolep** [influxdb/influxdb#4328]
During 200K EPS burn-in testing of TSM1, the follow panic brought testing to a halt.

```
found next stack barrier at 0xc8201a6f18; expected [*0xc820419f08=0x868f0b *0xc8201a6f80=0x461f01]                                                                                                                                                              
fatal error: missed stack barrier                                                                                                                                                                                                                               
                                                                                                                                                                                                                                                                
goroutine 0 [idle]:                                                                                                                                                                                                                                             
runtime.throw(0xcb0430, 0x14)                                                                                                                                                                                                                                   
        /home/philip/.gvm/gos/go1.5/src/runtime/panic.go:527 +0x90                                                                                                                                                                                              
runtime.gentraceback(0x41385f, 0x7f503f797b48, 0x0, 0xc8200f8180, 0x0, 0xc820101738, 0x40, 0x0, 0x0, 0x6, ...)                                                                                                                                                  
        /home/philip/.gvm/gos/go1.5/src/runtime/traceback.go:259 +0x1053                                                                                                                                                                                        
runtime.sigprof(0x41385f, 0x7f503f797b48, 0x0, 0xc8200f8180, 0xc8200f6000)                                                                                                                                                                                      
        /home/philip/.gvm/gos/go1.5/src/runtime/proc1.go:2585 +0x50b                                                                                                                                                                                            
runtime.sighandler(0xc80000001b, 0xc820101bb0, 0xc820101a80, 0xc8200f8180)                                                                                                                                                                                      
        /home/philip/.gvm/gos/go1.5/src/runtime/signal_amd64x.go:47 +0xbe                                                                                                                                                                                       
runtime.sigtrampgo(0x1b, 0xc820101bb0, 0xc820101a80)                                                                                                                                                                                                            
        /home/philip/.gvm/gos/go1.5/src/runtime/signal_linux.go:94 +0x95                                                                                                                                                                                        
runtime.sigtramp(0x1, 0x0, 0xc8200fa000, 0x0, 0x7fa0, 0x4000000, 0xc82dd21980, 0x1a74b6, 0x0, 0x15, ...)                                                                                                                                                        
        /home/philip/.gvm/gos/go1.5/src/runtime/sys_linux_amd64.s:234 +0x1b                                                                                                                                                                                     
runtime.sigreturn(0x0, 0xc8200fa000, 0x0, 0x7fa0, 0x4000000, 0xc82dd21980, 0x1a74b6, 0x0, 0x15, 0xd77cc8, ...)                                                                                                                                                  
        /home/philip/.gvm/gos/go1.5/src/runtime/sys_linux_amd64.s:238                                                                                                                                                                                           
                                                                                                                                                                                                                                                                
goroutine 3091302 [copystack]:                                                                                                                                                                                                                                  
fmt.(*pp).doPrintf(0xc8220212b0, 0xc24ba0, 0x7, 0xc820419630, 0x2, 0x2)                                                                                                                                                                                         
        /home/philip/.gvm/gos/go1.5/src/fmt/print.go:1082 fp=0xc8201a6538 sp=0xc8201a6530                                                                                                                                                                       
fmt.Sprintf(0xc24ba0, 0x7, 0xc820419630, 0x2, 0x2, 0x0, 0x0)                                                                                                                                                                                                    
        /home/philip/.gvm/gos/go1.5/src/fmt/print.go:203 +0x6f fp=0xc8201a6588 sp=0xc8201a6538                                                                                                                                                                  
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*Engine).nextFileName(0xc834ce3680, 0x0, 0x0)                                                                                                                                                                    
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/tsm1.go:1235 +0x1ad fp=0xc8201a6668 sp=0xc8201a6588                                                                                                                       
found next stack barrier at 0xc8201a6f18; expected [*0xc820419f08=0x868f0b *0xc8201a6f80=0x461f01]                                                                                                                                                              
fatal error: missed stack barrier
panic during panic

goroutine 0 [idle]:
runtime.startpanic_m()
        /home/philip/.gvm/gos/go1.5/src/runtime/panic1.go:67 +0x141
runtime.systemstack(0xd7c550)
        /home/philip/.gvm/gos/go1.5/src/runtime/asm_amd64.s:278 +0xab
runtime.startpanic()
        /home/philip/.gvm/gos/go1.5/src/runtime/panic.go:505 +0x14
runtime.throw(0xcb0430, 0x14)
        /home/philip/.gvm/gos/go1.5/src/runtime/panic.go:526 +0x83
runtime.gentraceback(0x477bc0, 0xc8201a6530, 0x0, 0xc82dd21980, 0x0, 0x0, 0x64, 0x0, 0x0, 0x0, ...)
        /home/philip/.gvm/gos/go1.5/src/runtime/traceback.go:259 +0x1053
runtime.traceback1(0xffffffffffffffff, 0xffffffffffffffff, 0x0, 0xc82dd21980, 0x0)
        /home/philip/.gvm/gos/go1.5/src/runtime/traceback.go:550 +0xc8
runtime.traceback(0xffffffffffffffff, 0xffffffffffffffff, 0x0, 0xc82dd21980)
        /home/philip/.gvm/gos/go1.5/src/runtime/traceback.go:527 +0x48
runtime.tracebackothers(0xc8200f8000)
        /home/philip/.gvm/gos/go1.5/src/runtime/traceback.go:664 +0xda
runtime.dopanic_m(0xc8200f8000, 0x42e830, 0xc820101588)
        /home/philip/.gvm/gos/go1.5/src/runtime/panic1.go:104 +0x1f9
runtime.dopanic.func1()
        /home/philip/.gvm/gos/go1.5/src/runtime/panic.go:514 +0x32
runtime.systemstack(0xc820101560)
        /home/philip/.gvm/gos/go1.5/src/runtime/asm_amd64.s:278 +0xab
runtime.dopanic(0x0)
        /home/philip/.gvm/gos/go1.5/src/runtime/panic.go:515 +0x61
runtime.throw(0xcb0430, 0x14)
        /home/philip/.gvm/gos/go1.5/src/runtime/panic.go:527 +0x90
runtime.gentraceback(0x41385f, 0x7f503f797b48, 0x0, 0xc8200f8180, 0x0, 0xc820101738, 0x40, 0x0, 0x0, 0x6, ...)
        /home/philip/.gvm/gos/go1.5/src/runtime/traceback.go:259 +0x1053
runtime.sigprof(0x41385f, 0x7f503f797b48, 0x0, 0xc8200f8180, 0xc8200f6000)
        /home/philip/.gvm/gos/go1.5/src/runtime/proc1.go:2585 +0x50b
runtime.sighandler(0xc80000001b, 0xc820101bb0, 0xc820101a80, 0xc8200f8180)
        /home/philip/.gvm/gos/go1.5/src/runtime/signal_amd64x.go:47 +0xbe
runtime.sigtrampgo(0x1b, 0xc820101bb0, 0xc820101a80)
        /home/philip/.gvm/gos/go1.5/src/runtime/signal_linux.go:94 +0x95
runtime.sigtramp(0x1, 0x0, 0xc8200fa000, 0x0, 0x7fa0, 0x4000000, 0xc82dd21980, 0x1a74b6, 0x0, 0x15, ...)
        /home/philip/.gvm/gos/go1.5/src/runtime/sys_linux_amd64.s:234 +0x1b
runtime.sigreturn(0x0, 0xc8200fa000, 0x0, 0x7fa0, 0x4000000, 0xc82dd21980, 0x1a74b6, 0x0, 0x15, 0xd77cc8, ...)
        /home/philip/.gvm/gos/go1.5/src/runtime/sys_linux_amd64.s:238
```

cc @jwilder @pauldix

**Comment by jwilder** [comment]
Possibly related: https://github.com/influxdb/influxdb/pull/4308#issuecomment-145632747

**Comment by dgnorton** [comment]
This issue in Go 1.5 looks similar:  https://github.com/golang/go/issues/12238.  A fix went into 1.5 but looks like they weren't able to exactly repro the issue.  Maybe there is still a related bug.

**Comment by otoolep** [comment]
I can bump our build process to 1.5.1 in an instant.

**Comment by otoolep** [comment]
Just say the word.

https://github.com/influxdb/influxdb/pull/4330

**Comment by otoolep** [comment]
https://github.com/golang/go/issues/12238#issuecomment-133234103

"As you can see from the stack trace, cgo code has called into Go code, the
Go code is allocating memory, the profiler has decided to jump in and get a
stack trace, and getting the stack trace crashes."

The testing that brought this out was running with profiling enabled.

**Comment by otoolep** [comment]
https://github.com/influxdb/influxdb/pull/4330

**Comment by otoolep** [comment]
@mjdesa hit this a second time, again with a binary built using 1.5.

**Comment by mjdesa** [comment]
@otoolep hit it another time with bc569e82debd611c38bb9d749b067af92a17b50c using go1.5.1

**Comment by DanielMorsing** [comment]
OK, I haven't seen anyone hit this on the go project bug tracker, so I think it's up to us to report it. @mjdesa, could you send me the logs you have so I can prepare something for the go folks?

**Comment by otoolep** [comment]
@mjdesa -- your test system is running with profiling enabled, correct?

**Comment by mjdesa** [comment]
correct

**Comment by mjdesa** [comment]
@otoolep I don't think the config option for profiling does anything.

```
[http]
  enabled = true
  bind-address = ":8086"
  auth-enabled = false
  log-enabled = false
  write-tracing = false
  pprof-enabled = false
```

```
2015/10/20 16:57:21 InfluxDB starting, version 0.9.5-nightly-7870b84, branch master, commit 7870b84f20e4164d8c94d37ae1566dfbd492cda5, built '2015-10-15T04:00:42+0000'
2015/10/20 16:57:21 Go version go1.5.1, GOMAXPROCS set to 32
2015/10/20 16:57:21 Using configuration at: /etc/opt/influxdb/influxdb.conf
2015/10/20 16:57:22 writing CPU profile to: /tmp/cpu.prof
2015/10/20 16:57:22 writing mem profile to: /tmp/mem.prof
```

**Comment by otoolep** [comment]
That config option in the file is a different thing, it does not control
the pprof shown in the log.

You'll see profiling has been enabled if you check influxd in the process
table.

**Comment by mjdesa** [comment]
@otoolep how do I shut it off?

I believe that I used the ansible playbook that had profiling disabled by default. I could be mistaken though.

**Comment by mjdesa** [comment]
false alarm.

Profiling was still enabled in `/etc/default/influxdb`.

**Comment by otoolep** [comment]
All testing so far shows that this is only an issue when profiling is enabled.

**Comment by pauldix** [comment]
I think this was fixed in 0.9.6
